extract-js-plugins.mjs

```diff
@@ -0,0 +1,44 @@
+import { readFile, writeFile } from "node:fs/promises";
+import { dirname, join } from "node:path";
+import { fileURLToPath } from "node:url";
+
+const workspaceDir = dirname(dirname(fileURLToPath(import.meta.url)));
```

Comment:
I would advise to write a comment section about what this script is doing and how.

Reply:
Yes you're totally right,

I changed the script to accept parameters to make testing easier, moved it to a `scripts` directory in which I put a README.md file (I found it more convenient for the explanation) but if you prefer I can add it as a comment at the top of the script.

Reply:
I prefer a comment instead of a separate file, this way it's improves the chances of someone trying to understand what the script is doing to actually find useful information, at least from my point of view.